src/wp-includes/interactivity-api/class-wp-interactivity-api.php

```diff
@@ -0,0 +1,959 @@
+		$new_style_property   = $style_property_value ? $style_property_name . ':' . $style_property_value . ';' : '';
+
+		// Generates an array with all the properties but the modified one.
+		foreach ( $style_assignments as $style_assignment ) {
+			if ( empty( trim( $style_assignment ) ) ) {
+				continue;
+			}
+			list( $name, $value ) = explode( ':', $style_assignment );
+			if ( trim( $name ) !== $style_property_name ) {
+				$result[] = trim( $name ) . ':' . trim( $value ) . ';';
+			}
+		}
+
+		// Adds the new/modified property at the end of the list.
+		$result[] = $new_style_property;
+
+		return implode( '', $result );
+	}
+
+	/**
+	 * Processes the `data-wp-text` directive.
+	 *
+	 * It updates the inner content of the current HTML element based on the
+	 * evaluation of its associated reference.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @param WP_Interactivity_API_Directives_Processor $p               The directives processor instance.
+	 * @param array                                     $context_stack   The reference to the context stack.
+	 * @param array                                     $namespace_stack The reference to the store namespace stack.
+	 */
+	private function data_wp_text_processor( WP_Interactivity_API_Directives_Processor $p, array &$context_stack, array &$namespace_stack ) {
+		if ( ! $p->is_tag_closer() ) {
+			$attribute_value = $p->get_attribute( 'data-wp-text' );
+			$result          = $this->evaluate( $attribute_value, end( $namespace_stack ), end( $context_stack ) );
+
+			/*
+			 * Follows the same logic as Preact in the client and only changes the
+			 * content if the value is a string or a number. Otherwise, it removes the
+			 * content.
+			 */
+			if ( is_string( $result ) || is_numeric( $result ) ) {
+				$p->set_content_between_balanced_tags( esc_html( $result ) );
+			} else {
+				$p->set_content_between_balanced_tags( '' );
+			}
+		}
+	}
+
+	/**
+	 * Returns the CSS styles for animating the top loading bar in the router.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @return string The CSS styles for the router's top loading bar animation.
+	 */
+	private function router_animation_styles(): string {
+		return <<<CSS
+			.wp-interactivity-router-loading-bar {
+				position: fixed;
+				top: 0;
+				left: 0;
+				margin: 0;
+				padding: 0;
+				width: 100vw;
+				max-width: 100vw !important;
+				height: 4px;
+				background-color: var(--wp--preset--color--primary, #000);
+				opacity: 0
+			}
+			.wp-interactivity-router-loading-bar.start-animation {
+				animation: wp-interactivity-router-loading-bar-start-animation 30s cubic-bezier(0.03, 0.5, 0, 1) forwards
+			}
+			.wp-interactivity-router-loading-bar.finish-animation {
+				animation: wp-interactivity-router-loading-bar-finish-animation 300ms ease-in
+			}
+			@keyframes wp-interactivity-router-loading-bar-start-animation {
+				0% { transform: scaleX(0); transform-origin: 0% 0%; opacity: 1 }
+				100% { transform: scaleX(1); transform-origin: 0% 0%; opacity: 1 }
```

Comment:
As per the [CSS coding standards](https://developer.wordpress.org/coding-standards/wordpress-coding-standards/css/), 0 values should not have units unless necessary, such as with transition-duration.

```suggestion
				0% { transform: scaleX(0); transform-origin: 0 0; opacity: 1 }
				100% { transform: scaleX(1); transform-origin: 0 0; opacity: 1 }
```